src/progressbar/bar.component.ts

```diff
@@ -30,7 +30,15 @@ import { isBs3 } from 'ngx-bootstrap/utils';
   }
 })
 export class BarComponent implements OnInit, OnDestroy {
-  max: number;
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+
+  set max(v: number) {
+    this._max = v;
+    this.applyTypeClasses();
```

Comment:
please move this to `OnChange ` and react only on type\max changes

Reply:
done